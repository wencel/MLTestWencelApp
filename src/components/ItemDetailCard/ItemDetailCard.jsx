import PropTypes from 'prop-types';
import Styles from './ItemDetailCard.module.sass';
import Loading from 'components/Loading';
import CurrencyFormat from 'react-currency-format';
import Button from 'components/Button';

/**
 * Component to render an item card, it contains error state and loading state.
 *
 * @component
 * @example
 * const item = {
      id: 'id1',
      title: 'item Title',
      price: {
        currency: 'ARS',
        amount: 29999,
        decimals: '00',
      },
      picture: 'pic url',
      condition: 'new',
      free_shipping: true,
      sold_quantity: 25,
      description: 'Item desc',
    },
  }
 * return (
 *   <ItemDetailCard item={item} loading={true} error=''/>
 * )
 */
const ItemDetailCard = ({ item, loading, error }) => {
  return (
    <>
      {!error && !loading && (
        <div className={Styles.ItemDetailCard}>
          <div className={Styles.imgContainer}>
            <img src={item.picture} alt='item img' />
          </div>
          <div className={Styles.infoContainer}>
            <div className={Styles.conditionContainer}>
              {item.condition === 'new' && 'Nuevo'}{' '}
              {item.condition === 'used' && 'Usado'} - {item.sold_quantity}{' '}
              vendidos
            </div>
            <div className={Styles.titleContainer}>{item.title}</div>
            <div className={Styles.priceContainer}>
              <CurrencyFormat
                value={item.price?.amount}
                displayType={'text'}
                thousandSeparator={true}
                prefix={'$'}
              />
              <div className={Styles.decimalsContainer}>
                {item.price?.decimals}
              </div>
            </div>
            <div className={Styles.buttonContainer}>
              <Button theme='blueTheme'>Comprar</Button>
            </div>
          </div>
          <div>
            <div className={Styles.descriptionTitleContainer}>
              Descripci√≥n del producto
            </div>
            <div className={Styles.descriptionContainer}>
              {item.description}
            </div>
          </div>
        </div>
      )}
      {loading && <Loading />}
      {error && <div>{error}</div>}
    </>
  );
};

ItemDetailCard.propTypes = {
  /**
   * error to show
   */
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  /**
   * Information about the item to be shown
   */
  item: PropTypes.shape({
    condition: PropTypes.string,
    description: PropTypes.string,
    picture: PropTypes.string,
    price: PropTypes.shape({
      amount: PropTypes.number,
      decimals: PropTypes.string,
    }),
    sold_quantity: PropTypes.number,
    title: PropTypes.any,
  }),
  /**
   * Loading flag
   */
  loading: PropTypes.bool,
};

export default ItemDetailCard;
