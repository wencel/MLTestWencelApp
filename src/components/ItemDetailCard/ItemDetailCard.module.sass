@import  'styles.global.sass'
.ItemDetailCard
  display: grid
  grid-template-columns: auto 20em
  padding: 2em
  grid-gap: 3em
  .imgContainer
    img
      width: 100%
  .conditionContainer
    font-size: 0.8em
  .titleContainer
    font-size: 1.5em
    font-weight: 500
    margin-top: 0.3em
  .priceContainer
    display: flex
    font-size: 2.3em
    line-height: 1
    margin-top: 0.5em
    margin-bottom: 1em
    .decimalsContainer
      font-size: 0.5em
      line-height: 1.2
  .descriptionTitleContainer
    font-size: 1.6em
    margin-bottom: 1em
  .descriptionContainer
    font-size: 0.85em
    line-height: 1.4
    color: $letterColorLight