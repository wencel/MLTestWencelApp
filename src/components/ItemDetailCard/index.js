import ItemDetailCard from './ItemDetailCard';
export default ItemDetailCard;
