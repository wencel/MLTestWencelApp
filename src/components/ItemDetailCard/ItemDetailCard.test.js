import React from 'react';
import { shallow } from 'enzyme';
import ItemDetailCard from './ItemDetailCard';
import Styles from './ItemDetailCard.module.sass';

describe('ItemDetailCard', () => {
  const item = {
    author: {
      name: 'Wencel',
      lastname: 'Santos',
    },
    item: {
      id: 'MLA901307677',
      title: 'Celular Tcl 10 Se Icy Silver',
      price: {
        currency: 'ARS',
        amount: 29999,
        decimals: '00',
      },
      picture: 'http://http2.mlstatic.com/D_690280-MLA44569794262_012021-I.jpg',
      condition: 'new',
      free_shipping: true,
      sold_quantity: 25,
      description:
        'IMPORTANTE: ¿Hacen factura A? Para que podamos emitir la factura A, necesitamos que la cuenta de Mercado Libre desde donde vayas a realizar la compra, esté registrada como IVA Responsable Inscripto o IVA Exento. De lo contrario, se va a emitir automáticamente una factura B. Podrás actualizar esta información desde la sección "Mis datos". \n\nSmartphone liberado con pantalla 20:9 V-notch HD+ de 6.5” (720x1600p) con tecnología NXTVISION, Procesador Octa Core, Android 10, Triple Cámara Principal 48+5+2mpx con dual LED flash - Frontal 13mpx/flash, Memoria interna 128 GB, memoria RAM 4 GB, Batería 4000mAh. Desbloqueo por huella y reconocimiento facial. Botón de acceso directo a Google Assistant. Incluye funda protectora.',
    },
  };
  let error = '';
  let loading = false;
  it('renders correctly with valid data', () => {
    const wrapper = shallow(
      <ItemDetailCard item={item} loading={loading} error={error} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(`.${Styles.ItemDetailCard}`).length).toEqual(1);
  });

  it('renders correctly when loading is taking place', () => {
    loading = true;
    const wrapper = shallow(
      <ItemDetailCard item={item} loading={loading} error={error} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(`.${Styles.ItemDetailCard}`).length).toEqual(0);
    expect(wrapper.find('Loading').length).toEqual(1);
  });

  it('renders correctly when an error has ocurred', () => {
    error = 'this is an error';
    loading = false;
    const wrapper = shallow(
      <ItemDetailCard item={item} loading={loading} error={error} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(`.${Styles.ItemDetailCard}`).length).toEqual(0);
  });
});
