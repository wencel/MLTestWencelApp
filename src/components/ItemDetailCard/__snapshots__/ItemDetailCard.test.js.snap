// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetailCard renders correctly when an error has ocurred 1`] = `ShallowWrapper {}`;

exports[`ItemDetailCard renders correctly when loading is taking place 1`] = `ShallowWrapper {}`;

exports[`ItemDetailCard renders correctly with valid data 1`] = `ShallowWrapper {}`;
